### PR TITLE
Feat/children html pdf

### PR DESCRIPTION
**Pull request by dannil76**
## Feature description
Added feature to render PDF from form answers with information about children

## Solution description
Added new helper function 'createChildren' executed by 'createRecurringCaseTemplate' and return an list (children) of children based on form answers.
The children list is then used in the handlebars template 

## What areas is affected by these changes?.
Viva-ms
Html-pdf-ms

## Was this feature tested in the following environments?
- [x] Your personal AWS environment.

### CODE REVIEW

**Comment by sebastianthulin** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/helsingborg-stad/repositories/602c13c73b93a7537d767e421a43afaa/pull-request/264)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1jfxpqc Lägg till barn information från ärende i mall-data](https://app.clickup.com/t/1jfxpqc)

**Comment by helsingborgsstad** [comment]
Task linked: [CU-1jfxpuk Uppdatera mall så att personinformation om ett, flera eller inga barn visas korrekt.](https://app.clickup.com/t/1jfxpuk)

File: services/viva-ms/helpers/formHelpers.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ function getAttributeFromAnswerFieldId(fieldId) {
   return attribute;
 }
 
+function getSecondStringFromDotNotatedString(sourceString) {
```
Comment (TeddyAndersson) [inline services/viva-ms/helpers/formHelpers.js:15]: If this function is to live in the form-helpers lib. It should be renamed to something more accurate and repurposed.

```JS
getAttributeFromDotNotation(string, position) {
  const attributes = string.split('.');
    return attributes[position];
}
```

File: services/viva-ms/lambdas/generateRecurringCaseHtml.js [hunk 2]
```diff
@@ -119,19 +122,24 @@ export async function main(event, context) {
   return true;
 }
 
-function updateVivaCaseState(caseKeys) {
+function isCaseHtmlGenerated(state) {
+  return String(state).includes(CASE_HTML_GENERATED);
```
[the remark this answers is not in the document]
Comment (dannil76) [reply services/viva-ms/lambdas/generateRecurringCaseHtml.js:126]: Will revert

File: services/viva-ms/lambdas/generateRecurringCaseHtml.js [hunk 3]
```diff
@@ -37,6 +37,16 @@ export async function main(event, context) {
...
 
+  if (isCaseHtmlGenerated()) {
+    log.info(
+      `Html is already generated for case with id: ${caseItem.id}`,
+      context.awsRequestId,
+      'service-viva-ms-generateRecurringCaseHtml-000',
+      caseItem.state
+    );
+    return true;
+  }
```
[the remark this answers is not in the document]
Comment (dannil76) [reply services/viva-ms/lambdas/generateRecurringCaseHtml.js:48]: Will revert

**Review by TeddyAndersson — APPROVED** [review]
Ship it!

**Pull request merged**